File: @LV1/1.4/babybites/babybites.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var n int
	fmt.Scanf("%d", &n)

	var s string
	cnt := 1
	isMakeSense := true

	for i := 0; i < n; i++ {
		fmt.Scanf("%s", &s)

		num, err := strconv.Atoi(s)

		if err == nil && num != cnt {
			isMakeSense = false
			break
		}

		cnt++
	}

	if isMakeSense {
		fmt.Println("makes sense")
	} else {
		fmt.Println("something is fishy")
	}
}

File: LV1/1.4/babybites/babybites.hs
import Data.Char

isMakeSense :: [String] -> Int -> Bool
isMakeSense (x : xs) cnt = if isDigit $ head x
  then (read x == cnt) && isMakeSense xs (cnt + 1)
  else isMakeSense xs (cnt + 1)
isMakeSense _ _ = True

main :: IO ()
main = do
  _ <- getLine
  l <- words <$> getLine
  putStrLn $ if isMakeSense l 1
    then "makes sense"
    else "something is fishy"
